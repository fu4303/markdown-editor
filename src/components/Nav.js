import React from "react"

const Nav = () => (
  <nav>
    <h1>Markdown Forge</h1>
    <ul>
      <li>Info</li>
      <li>Github</li>
      <li>Twitter</li>
    </ul>
  </nav>
)

export default Nav
