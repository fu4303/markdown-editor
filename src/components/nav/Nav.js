import React from "react"
import PropTypes from "prop-types"
import { Link, Match } from "@reach/router"
import Slide from "react-reveal/Slide"
import { useOvermind } from "../../hooks/useOvermind"
import { NavWrapper, NavContent, NavLogo, NavMenu, NavMenuItem } from "./styles"
import Button from "../button"

const Nav = () => {
  const {
    actions: { saveMarkdown, resetMarkdown },
  } = useOvermind()

  return (
    <Slide top delay={250} duration={750}>
      <NavWrapper>
        <NavContent>
          <NavLogo>
            <Link to="/">
              <span>Markdown</span> Forge
            </Link>
          </NavLogo>
          <NavMenu>
            <NavMenuItem>
              <Link to="/about">About</Link>
            </NavMenuItem>
            <Match path="/">
              {(props) =>
                props.match && (
                  <>
                    <NavMenuItem>
                      <Button
                        type="button"
                        action={resetMarkdown}
                        text="Reset"
                      />
                    </NavMenuItem>
                    <NavMenuItem>
                      <Button
                        type="button"
                        action={saveMarkdown}
                        text="Download"
                      />
                    </NavMenuItem>
                  </>
                )
              }
            </Match>
          </NavMenu>
        </NavContent>
      </NavWrapper>
    </Slide>
  )
}

Nav.propTypes = {
  // eslint-disable-next-line react/forbid-prop-types
  match: PropTypes.object,
}

Nav.defaultProps = {
  match: {},
}

export default Nav
