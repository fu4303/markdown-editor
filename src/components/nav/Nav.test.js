import React from "react"
import { shallow } from "enzyme"
import toJson from "enzyme-to-json"
import Nav from "./Nav"

describe("<Nav />", () => {
  it("renders without crashing", () => {
    shallow(<Nav />)
  })

  it("matches snapshot with data passed correctly", () => {
    const wrapper = shallow(<Nav />)
    expect(toJson(wrapper)).toMatchSnapshot()
  })
})
