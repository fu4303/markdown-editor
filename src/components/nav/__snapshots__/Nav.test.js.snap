// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<Nav /> matches snapshot with context passed correctly 1`] = `<Nav />`;
